tests/test_java.py

```diff
@@ -653,3 +658,46 @@ def test_java_different_basename(
                     os.path.basename(log_records[0].gprofiler_adapter_extra["exe"])  # type: ignore
                     == java_notjava_basename
                 )
+
+
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("insert_dso_name", [False, True])
+def test_dso_name_in_ap_profile(
+    tmp_path: Path,
+    application_pid: int,
+    insert_dso_name: bool,
+) -> None:
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        insert_dso_name=insert_dso_name,
+        duration=3,
+        frequency=999,
+    ) as profiler:
+        collapsed = snapshot_pid_profile(profiler, application_pid).stacks
+        assert is_function_in_collapsed("jni_NewObject", collapsed)
+        assert insert_dso_name == is_pattern_in_collapsed(r"jni_NewObject \(.+?/libjvm.so\)", collapsed)
+
+
+# test that occurrence of raw DSO path as a symbol is recognized as such by async profiler
```

Comment:
The idea here is specifically a symbol that's missing so we will get an unknown?

Reply:
Yes, that's the similar case.
More specifically I'm referring to the following sample profile: https://github.com/Granulate/async-profiler/pull/2#issue-1385783041